lib/services/spam-detection.js

```diff
@@ -113,6 +115,15 @@ class SpamDetection {
     return this.isSimilarityAboveThreshold(matchPercents);
   }
 
+  checkListOfRecentUrlsForSpam(message, urlList) {
+    return this.messageMatching.getLinks(message).some((link) => {
+      return (
+        urlList.filter((url) => url === link.hostname + link.pathname).length >=
+        this.messageUrlSpamCount - 1
```

Comment:
This will trigger the mute after the link is sent only the second time, right? Is that intentional?

Reply:
It will trigger on the 3rd (messageUrlSpamCount) link message because because the message gets spam checked before being added to messages cache.